arangod/IResearch/IResearchView.cpp

```diff
@@ -1613,21 +1622,29 @@ arangodb::Result IResearchView::link(
   dataPath /= subPath;
   dataPath /= arangodb::iresearch::IResearchView::type().name();
   dataPath += "-";
-  dataPath += std::to_string(view->id());
+  dataPath += std::to_string(id);
+
+  auto view = std::unique_ptr<IResearchView>(
```

Comment:
Should we use `std::make_unique<>` here?

Reply:
unfortunately `std::make_unique` as well as `std::make_shared` are unable to create object with private constructors. If you know workaround for that issue would be great to learn it.

Reply:
There's not a way to shoehorn a private constructor into `make_unique`, but there are some workarounds. For instance, you can make the constructor public, but give it a parameter with a type that is a private member of the class, then make a static factory method on the class, e.g. `IResearchView::Create` which uses `make_unique` and the constructor guard parameter. I believe I did something like this in `arangod/Cache/TransactionalCache.cpp` (method was `create`). It's unclear how much benefit this provides, as it primarily allows you to use `make_unique` or `make_shared`, but leads to a more convoluted pattern.

Reply:
@danielhlarkin thank you for your note. I'm aware about this trick but I personally don't like it))
there is another one I forgot about:
https://gist.github.com/gnusi/7496e8e67b194a9a56eebf5ba807fac0
Looks much nicer from my perspective but requires class/struct to be non `final`